pages/_app.tsx

```diff
@@ -74,11 +78,13 @@ function App({ Component, pageProps }) {
       />
       <SessionProvider refetchInterval={5 * 60} session={pageProps.session}>
         <QueryClientProvider client={queryClient}>
-          <StoreProvider>
-            <ChakraProvider theme={theme}>
-              <Component {...pageProps} />
-            </ChakraProvider>
-          </StoreProvider>
+          <Hydrate state={pageProps.dehydratedState}>

```

Comment:
What does this do?

Reply:
This takes the result and adds it statically to the output of the document at request time.

Reply:
Ok I need to read up on it - do we get any perf benefits just by adding it straight away? And FYI so far there is no explicit SSR, just static generation or fully dynamic (on the admin side).

But there's probably some automatic SSR that's happening by default.

Reply:
So it's the difference between there being a first render without any buttons and one with. That said it does mean the core route is now an SSR route with static routes under it. I'm not tied to using SSR for this, I just didn't have a loading state for the client rendered version.